### PR TITLE
Use tippy.js for context popup

### PR DESCRIPTION
**Pull request by silverwind**
By appending the tooltips to `document.body`, we can avoid any stacking context issues caused by surrounding element's CSS.

I wasn't able to get this portalling to work with Fomantic popups, so this uses [tippy.js](https://github.com/atomiks/tippyjs) instead of Fomantic popups. We should aim to replace all Fomantic popups with this eventually and then get rid of the Fomantic `popup` module completely.

Fixes: https://github.com/go-gitea/gitea/pull/20377

image

### CODE REVIEW

**Comment by silverwind** [comment]
The pop-in animation is changed. Fomantic used a scale-in origination from the cursor point, which I wasn't able to replicate with tippy's "scale" animation which would scale from the center and wouldn't look so good. So I left it at tippy's default "fade" animation.

**Comment by wxiaoguang** [comment]
If the new tippy.js is introduced, will the Fomantic-UI popup be removed? To avoid potential arguments about using tippy or popup for new PRs in the future. If both of them exist, it seems not clear about which one should be used.

**Comment by silverwind** [comment]
Complete removal of Fomantic `popup` module will be a much bigger effort. I'd like to take it step-by-step and this is the first feature that will use tippy.

File: web_src/less/modules/tippy.less [hunk 1]
```diff
@@ -0,0 +1,82 @@
+.tippy-box[data-animation="fade"][data-state="hidden"] {
```
Comment (Gusted) [inline web_src/less/modules/tippy.less:3]: Should this only happen when `[data-animation="fade"]`, seems like it should be independent of the animation.

Comment (silverwind) [reply web_src/less/modules/tippy.less:3]: Some of these styles come from [tippy.js's suggested CSS](https://github.com/atomiks/tippyjs/blob/master/src/scss/animations/fade.scss), this is one of them.

Comment (silverwind) [reply web_src/less/modules/tippy.less:3]: In fact, this rule and the `transition-property` below enables the `fade` animation. Not that I like it but I guess we could just leave it in.

**Review by Gusted — APPROVED** [review]

**Comment by silverwind** [comment]
Did some investigation, luckily the Fomantic `popup` module is not being depended on by other fomantic modules, so I guess it shouldn't be too much work to replace all tooltips/popups with tippy.

File: web_src/less/modules/tippy.less (hunk 1 above)
Comment (wxiaoguang) [reply web_src/less/modules/tippy.less:3]: I think there could be more comments about where the styles come from and what they are for.

**Comment by wxiaoguang** [comment]
> Did some investigation, luckily the Fomantic `popup` module is not being depended on by other fomantic modules, so I guess it shouldn't be too much work to replace all tooltips/popups with tippy.
> 
> Still it'll probably be a few hours work to do it, so if you want to merge now, go ahead.

Good news, then LGTM (a small nit about comment above).

File: web_src/less/modules/tippy.less (hunk 1 above)
Comment (silverwind) [reply web_src/less/modules/tippy.less:3]: Comment on style source added.

**Review by 6543 — APPROVED** [review]

**Pull request merged**